api/src/main/java/jakarta/config/Config.java

```diff
@@ -118,6 +119,21 @@
      */
     <T> Optional<List<T>> asList(Class<T> type);
 
+    /**
+     * Register a change consumer for this config node. The consumer would only be called if a change in configuration impacts
+     * the current node or its sub-nodes.
+     * <p>
+     * Example 1: If this node is {@code server.host} and {@code server.host} is modified, the consumer would be triggered,
+     *  in case {@code server.port} is modified, the consumer would not be triggered.
+     * <p>
+     * Example 2: If this node is {@code server} and either {@code server.host} or {@code server.port} is modified, the
+     * consumer would be triggered.
+     *
+     * @param changeConsumer consumer of changes, the config provided to the consumer is the same node as the current node,
+     *                          but with values from the changed config source
+     */
+    void onChange(Consumer<Config> changeConsumer);
```

Comment:
We probably need a unsubscribe mechanism.

Reply:
I will change the method so the listener may return a boolean to mark whether it is still interested in changes. This may require one more event to be triggered, but is easier than handling subscriber identification.